### PR TITLE
nixosTests.nginx*: port to python

### PR DESCRIPTION
**Pull request by globin**
###### Motivation for this change
#72828 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/nginx.nix [hunk 1]
```diff
@@ -69,43 +69,46 @@ import ./make-test.nix ({ pkgs, ... }: {
...
+        assert httpCode == "304"
+
+        return etag
+
+
+    webserver.wait_for_unit("nginx")
+    webserver.wait_for_open_port("80")
+
+    with subtest("check ETag if serving Nix store paths"):
+        oldEtag = checkEtag
```
Comment (flokli) [inline nixos/tests/nginx.nix]: `checkEtag` is a function. I think this should be called
```py
old_etag = check_etag()
```

File: nixos/tests/nginx.nix [hunk 2]
```diff
@@ -69,43 +69,46 @@ import ./make-test.nix ({ pkgs, ... }: {
...
+        webserver.succeed(
+            "${etagSystem}/bin/switch-to-configuration test >&2"
+        )
+        webserver.sleep(1)  # race condition
```
Comment (flokli) [inline nixos/tests/nginx.nix]: What race condition happens here? We probably need to wait longer, just to be sure…

File: nixos/tests/nginx.nix [hunk 3]
```diff
@@ -69,43 +69,46 @@ import ./make-test.nix ({ pkgs, ... }: {
...
+        newEtag = checkEtag
```
Comment (flokli) [inline nixos/tests/nginx.nix]:
```suggestion
        new_etag = check_etag()
```

File: nixos/tests/nginx.nix [hunk 4]
```diff
@@ -69,43 +69,46 @@ import ./make-test.nix ({ pkgs, ... }: {
...
+        assert oldEtag != newEtag
```
Comment (flokli) [inline nixos/tests/nginx.nix]:
```suggestion
        assert old_etag != new_etag
```

File: nixos/tests/nginx.nix [hunk 5]
```diff
@@ -69,43 +69,46 @@ import ./make-test.nix ({ pkgs, ... }: {
...
+
+    with subtest("config is reloaded on nixos-rebuild switch"):
+        webserver.succeed(
+            "${justReloadSystem}/bin/switch-to-configuration test >&2"
+        )
+        webserver.wait_for_open_port("8080")
```
Comment (flokli) [inline nixos/tests/nginx.nix]:
```suggestion
        webserver.wait_for_open_port(8080)
```

**Review by flokli — CHANGES_REQUESTED** [review]

File: nixos/tests/nginx.nix (hunk 2 above)
Comment (globin) [reply nixos/tests/nginx.nix]: I've taken this out for now, as I can't reproduce the failure even on a build machine running over load.

File: nixos/tests/nginx-sso.nix [hunk 6]
```diff
@@ -27,18 +27,20 @@ import ./make-test.nix ({ pkgs, ... }: {
...
-    startAll;
+    start_all()
 
-    $machine->waitForUnit("nginx-sso.service");
-    $machine->waitForOpenPort(8080);
+    machine.wait_for_unit("nginx-sso.service")
+    machine.wait_for_open_port(8080)
 
     # No valid user -> 401.
-    $machine->fail("curl -sSf http://localhost:8080/auth");
```
Comment (tfc) [inline nixos/tests/nginx-sso.nix]: the comments could be the titles of `subtest`s so this information is even visible in the log

**Review by tfc — CHANGES_REQUESTED** [review]
Please make the comments with the http return codes subtest titles, that would be awesome.

**Comment by flokli** [comment]
ping @globin

**Comment by Ma27** [comment]
@GrahamcOfBorg test nginx nginx-sso

**Comment by Ma27** [comment]
Rebased onto latest master, fixed the `nginx` test.

@GrahamcOfBorg test nginx nginx-sso

File: nixos/tests/nginx-sso.nix (hunk 6 above)
Comment (Ma27) [reply nixos/tests/nginx-sso.nix]: Done :)

**Review by Ma27 — APPROVED** [review]

**Pull request merged**